tests/unittest.py

```diff
@@ -334,7 +334,7 @@ def wait_on_thread(self, deferred, timeout=10):
     def wait_for_background_updates(self) -> None:
         """Block until all background database updates have completed."""
         while not self.get_success(
-            self.store.db_pool.updates.has_completed_background_updates()
+            self.hs.get_datastore().db_pool.updates.has_completed_background_updates()
         ):
             self.get_success(
                 self.store.db_pool.updates.do_next_background_update(100), by=0.1
```

Comment:
I think you need to replace it here too, maybe just pull `store` out to a separate variable for this function?